middle_end/flambda/simplify/simplify_set_of_closures.rec.ml

```diff
@@ -146,7 +148,9 @@ end = struct
                   Code_id.Map.find (FD.code_id function_decl)
                     old_to_new_code_ids_all_sets
                 in
-                function_decl_type denv function_decl
+                function_decl_type
+                  ~pass:Before_simplify
```

Comment:
If you ditch your complicated module structure, this could become `~pass:Inlining_report.Before_simplify` which I think is not too long, clear, and doesn't rely on disambiguation through inferred types.

Reply:
done